### PR TITLE
Add missing reexport in logger

### PR DESCRIPTION
**Pull request by Ristarg**
Followup to #976. I tested the new functionality on the internal examples, so I didn't catch the missing visibility; sorry about that. Took this chance to also update the docs.

### CODE REVIEW

**Comment by torkleyy** [comment]
@jojolepro I don't think that should be a concern here. We've got modules for that, so you don't need to prefix stuff. You can always just import the module or rename the item when importing.

**Comment by Ristarg** [comment]
I think Amethyst could definitely utilize namespacing better than it does right now, instead of polluting the crate's namespace with reexports from private modules. But that should probably come in the context of a bigger PR.

**Comment by jojolepro** [comment]
Its a matter of preference I guess. I'll need to play with crossterm's Stdout and amethyst's logger Stdout in the same time when I integrate the terminal duplex

**Comment by torkleyy** [comment]
@jojolepro Yes, so using modules or renaming would be much better than hoping that random word additions avoid collisions. It's understandable in a single crate that clearly differentiable names are better (for the understanding, not the convenience of not renaming). But deciding to rename an item here (even making it longer) because it collides with other crates doesn't make that much sense. I mean, we even have items that collide with very common things (think of `Result`).

In short: Rename for the purpose of clarity, not to avoid collisions.

**Comment by jojolepro** [comment]
I'll change the logger in a few days so I'll see there what I can do for the name. Thanks!
bors r+
